lib/tests/formulae.rb

```diff
@@ -39,6 +41,106 @@ def run!(args:)
+          event_payload = JSON.parse(File.read(event_path))
+
+          before = event_payload.fetch("before", nil)
+          test git, "-C", repository, "fetch", "origin", before if before.present?
+
+          before
+        end
+
+        @previous_sha
+      end
+
+      GRAPHQL_QUERY = <<~GRAPHQL
+        query ($node_id: ID!) {
+          node(id: $node_id) {
+            ... on Commit {
+              checkSuites(last: 100) {
+                nodes {
+                  status
+                  workflowRun {
+                    databaseId
+                    event
+                    workflow {
+                      name
+                    }
+                  }
+                  checkRuns(last: 100) {
+                    nodes {
+                      name
+                      status
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      GRAPHQL
+
+      def download_previously_built_bottles!
+        return if previous_github_sha.blank?
+
+        repo = ENV.fetch("GITHUB_REPOSITORY")
+        url = GitHub.url_to("repos", repo, "commits", previous_github_sha)
+        response = GitHub::API.open_rest(url)
+        node_id = response.fetch("node_id")
+
+        response = GitHub::API.open_graphql(GRAPHQL_QUERY, variables: { node_id: node_id })
+        check_suite_nodes = response.dig("node", "checkSuites", "nodes")
+        return if check_suite_nodes.blank?
+
+        formulae_tests_node = check_suite_nodes.find do |node|
+          next false if node.fetch("status") != "COMPLETED"
+
+          workflow_run = node.fetch("workflowRun")
+          next false if workflow_run.fetch("event") != "pull_request"
+          next false if workflow_run.dig("workflow", "name") != "CI"
+
+          check_run_nodes = node.dig("checkRuns", "nodes")
+          next false if check_suite_nodes.blank?
+
+          check_run_nodes.any? do |check_run_node|
+            check_run_node.fetch("name") == "conclusion" && check_run_node.fetch("status") == "COMPLETED"
+          end
+        end
+        return if formulae_tests_node.blank?
+
+        run_id = formulae_tests_node.dig("workflowRun", "databaseId")
+        return if run_id.blank?
+
+        url = GitHub.url_to("repos", repo, "actions", "runs", run_id, "artifacts")
+        response = GitHub::API.open_rest(url)
+        return if response.fetch("total_count").zero?
+
+        bottles_artifact = response.fetch("artifacts").find { |artifact| artifact.fetch("name") == "bottles" }
+        return if bottles_artifact.blank?
+
+        ohai "Downloading bottles from #{previous_github_sha}"
+        download_url = bottles_artifact.fetch("archive_download_url")
+        GitHub.download_artifact(download_url, run_id)
+      end
+
+      def no_diff?(formula, sha)
+        relative_formula_path = formula.path.relative_path_from(repository)
+        system(git, "-C", repository, "diff", "--quiet", sha, relative_formula_path)
```

Comment:
```suggestion
        system(git, "-C", repository, "diff", "--no-ext-diff", "--quiet", sha, relative_formula_path)
```

just in case

Reply:
I suspect it does, but there's no mention in the docs and I'm not in the mood to look at the source right now, so will apply.